kaska/tests/test_smoothn.py

```diff
@@ -15,29 +15,52 @@
 
 # Make a test from the first example of the smoothn code
 def test_robust1d():
+    unrobust_target = 0.7503929639274534
+    robust_target = 0.18943656067148762
+
+    prec = 1e-5
+
+    general_robustness_test(isrobust=None,
+                            target=unrobust_target,
+                            s=None, prec=prec)
+    general_robustness_test(isrobust=True,
+                            target=robust_target,
+                            s=None, prec=prec)
+
+def general_robustness_test(isrobust=None, target=0.0, s=None, prec=1e-12):
     x = np.linspace(0, 100, 256)
     y_base = np.cos(x/10) + (x/50.)**2
     np.random.seed(3141592653)
     y_noise = y_base + np.random.randn(len(x))/10
     y = np.copy(y_noise) ; y[70:85:5] = (5.5, 5, 6)
-    (z, s, flag, wtot) = smoothn.smoothn(y)
-    (zr, sr, flag, wtot) = smoothn.smoothn(y, isrobust=True)
+    if (isrobust is None):
+        (z, s, flag, wtot) = smoothn.smoothn(y, s=s)
+    else:
+        (z, s, flag, wtot) = smoothn.smoothn(y, s=s, isrobust=isrobust)
+
     res = z - y_base
-    resr = zr - y_base
+
+    assert np.abs(np.max(np.abs(res)) - target) < prec
     
-    unrobust_target = 0.7503929639274534
-    robust_target = 0.18943656067148762
     
-    #prec = 1e-12
+def test_fixed_order():
+    unrobust_target = 0.7503929639274534
+    unrobust_order = 56.93236088601813
```

Comment:
What are these magic numbers in `test_fixed_order()` ?

Reply:
The parameters and results for the test? I can add a comment to that effect.